### PR TITLE
refactor: use Spectral to check for duplicate parameters

### PR DESCRIPTION
**Pull request by barrett-schonefeld**
Purpose:
- Simplify the validator ruleset by using an equivalent Spectral rule
- Avoid resolving spec in-place to ensure inCodeValidator does not modify user's original spec object

Changes:
- Enable the operation-parameters Spectral rule
- Remove the in-code validation for parameters with duplicate `in:` and `name:` properties
- Pass a copy of the original spec to the reference resolver to prevent modification of the original spec

Tests:
- Remove tests meant to ensure the operation-parameters test is disabled
- Remove invalid oas3 from `disable-rules.yml` meant to invoke operation-parameters errors
- Remove irrelevant tests for in-code validation
- Add test to ensure operation-parameters check catches duplicate parameters for oas3 document

### CODE REVIEW

File: test/spectral/mockFiles/swagger/enabled-rules.yml [hunk 1]
```diff
@@ -166,6 +199,19 @@ definitions:
...
+      properties:
+        pet_id:
+          description: pet_id
+          type: integer
+          format: int64
+          example: 10
+        name:
+          description: pet name
+          type: string
+          example: doggie
```
Comment (mkistler) [inline test/spectral/mockFiles/swagger/enabled-rules.yml:214]: This is incredibly picky but it will demonstrate that I _did_ look somewhat carefully.  I think the section I highlighted is indented one level more than it needs to be.

**Review by mkistler — APPROVED** [review]
Looks good! 👍

File: test/plugins/validation/oas3/responses.test.js [hunk 2]
```diff
@@ -645,7 +645,6 @@ describe('validation plugin - semantic - responses - oas3', function() {
     };
 
     const res = validate({ resolvedSpec: spec }, config);
-    console.log(res.warnings);
```
Comment (dpopp07) [inline test/plugins/validation/oas3/responses.test.js:648]: Good catch!

**Review by dpopp07 — APPROVED** [review]
Looks great! 👍

File: test/spectral/mockFiles/swagger/enabled-rules.yml (hunk 1 above)
Comment (barrett-schonefeld) [reply test/spectral/mockFiles/swagger/enabled-rules.yml:241]: Yes, nice catch!
